challenge-16/challenge-16.js

```diff
@@ -1,70 +1,91 @@
+	- "João, Maria, Roberto, Pedro e Marcos são meus amigos."
+	- Detalhe: o código abaixo deve funcionar com um array de qualquer tamanho.
+	5 nomes foi somente uma sugestão ;)
+	*/
+	console.log( '\nMeus amigos:' );
+	var friends = ['Joice', 'Jessica', 'Tati', 'Carol', 'Bia'];
+	var phrase = friends.reduce(function(acumulado, atual, index) {
+		var separator = friends.length - 1 === index ? ' e ' : ', ';
+		return acumulado + separator + atual;
+	}).concat( ' são meu amigos.' );
+	console.log( phrase );
 
-/*
-- Crie um array com 5 nomes. Reduza esses nomes a uma única string, separando
-cada nome por vírgula. Entre o penúltimo e o último nome, o separador deve
-ser um "e". No final, concatene com a frase: " são meus amigos.".
-O resultado final deve ficar mais ou menos assim:
-- "João, Maria, Roberto, Pedro e Marcos são meus amigos."
-- Detalhe: o código abaixo deve funcionar com um array de qualquer tamanho.
-5 nomes foi somente uma sugestão ;)
-*/
-console.log( '\nMeus amigos:' );
-// ?
+	/*
+	Usando o replace(), faça a string "Roberto" virar "Roberta".
+	Mostre o resultado no console.
+	*/
+	console.log( '\nEra "Roberto", agora é:' );
+	console.log( 'Roberto'.replace( 'to', 'ta' ) );
 
-/*
-Usando o replace(), faça a string "Roberto" virar "Roberta".
-Mostre o resultado no console.
-*/
-console.log( '\nEra "Roberto", agora é:' );
-// ?
+	/*
+	Mostre no console a parte "nando" da string "Fernando". Use o método que
+	faz a busca do final para o início da string.
+	*/
+	console.log( '\nParte de uma string:' );
+	console.log( 'Fernando'.substring( 8, 3 ) );
 
-/*
-Mostre no console a parte "nando" da string "Fernando". Use o método que
-faz a busca do final para o início da string.
-*/
-console.log( '\nParte de uma string:' );
-// ?
-
-/*
-Declare uma variável chamada `myName`, que receba o seu primeiro nome,
-escrito de forma natural.
-Mostre no console esse nome, mas com as letras intercalando entre maiúsculas
-e minúsculas.
-- Detalhe: o código abaixo deve funcionar da mesma forma para qualquer nome,
-de qualquer tamanho, escrito de qualquer forma.
-Ex.: Nomes que deveriam funcionar: "Fernando", "RoBertO", "gabriEla", etc.
-*/
-console.log( '\nNome com letras intercaladas entre caixa alta e baixa:' );
-// ?
\ No newline at end of file
+	/*
+	Declare uma variável chamada `myName`, que receba o seu primeiro nome,
+	escrito de forma natural.
+	Mostre no console esse nome, mas com as letras intercalando entre maiúsculas
+	e minúsculas.
+	- Detalhe: o código abaixo deve funcionar da mesma forma para qualquer nome,
+	de qualquer tamanho, escrito de qualquer forma.
+	Ex.: Nomes que deveriam funcionar: "Fernando", "RoBertO", "gabriEla", etc.
+	*/
+	console.log( '\nNome com letras intercaladas entre caixa alta e baixa:' );
+	var myName = 'Aline';
+	var myNewName = [];
+	for( var i = 0, len = myName.length; i < len; i++ ) {
+		myNewName.push( i % 2 === 0 ? myName[i].toLowerCase() : myName[i].toUpperCase() );
+	}.join( '' );
```

Comment:
-1
O `for` não retorna nada, por isso esse código gera um erro. O `for` é só um bloco de repetição. Os métodos de array (map, filter, reduce, etc), retornam valores, então você consegue fazer isso que você tentou :)